server/app/views/components/ButtonUtils.java

```diff
@@ -0,0 +1,49 @@
+package views.components;
+
+import static j2html.TagCreator.button;
+import static j2html.TagCreator.form;
+import static j2html.TagCreator.input;
+import static j2html.TagCreator.span;
+
+import j2html.tags.ContainerTag;
+import j2html.tags.Tag;
+import org.apache.commons.lang3.RandomStringUtils;
+import play.filters.csrf.CSRF;
+import play.mvc.Http;
+import scala.Option;
+import views.style.Styles;
+
+/** Contains helpers for creating buttons for use in the UI. */
+public final class ButtonUtils {
```

Comment:
issue (blocking): the views already have two classes that contain logic for creating different types of buttons, I'd prefer to refactor toward having one vs introducing a third

Reply:
Discussed offline and agreed. My main goal is to reduce the number of ways we attempt to wrap creation of the underlying element. In my opinion, that pattern doesn't grow extremely well and eventually enough customizations are added to it (to support more scenarios) that it ends up being an almost-complete set of the functionality offered by the original element it was wrapping. For now, we were planning on the following:

1. Modify current PR so that everything in `ButtonUtils` moves to `BaseHtmlView`. Where appropriate (e.g. redirectButton), modify existing callers to pass in a constructed button element that the method augments (rather than constructing a new button). Don’t touch existing `BaseHtmlView.button()` helpers.
2. Post PR-submit, do a pass to remove unnecessary button methods from `BaseHtmlView`.
3. As more views are converted, move away from the `LinkElement` pattern in favor of using the helpers in `BaseHtmlView`.
4. If we find the number of helpers in `BaseHtmlView` to be unwieldy in the future, move out to a static helper class